stm32/libcanard/canard_stm32.c

```diff
@@ -11,10 +11,15 @@
 #include <unistd.h>
 
 
-#if CANARD_STM32_USE_CAN2
+#if CANARD_STM32_USE_CAN3
+# define BXCAN                                                  CANARD_STM32_CAN3
+# define FILTER_CONFIG_BXCAN                                    CANARD_STM32_CAN3
+#elif CANARD_STM32_USE_CAN2
 # define BXCAN                                                  CANARD_STM32_CAN2
-#else
+# define FILTER_CONFIG_BXCAN                                    CANARD_STM32_CAN1
+#elif CANARD_STM32_USE_CAN1
 # define BXCAN                                                  CANARD_STM32_CAN1
+# define FILTER_CONFIG_BXCAN                                    CANARD_STM32_CAN1
 #endif
```

Comment:
I suggest we add an `ifndef` check below that makes sure that BXCAN is defined. If not, the user forgot to define any of `CANARD_STM32_USE_CAN*`.